lib/awskeyring.rb

```diff
@@ -26,8 +26,8 @@ def self.init_keychain(awskeyring:)
   end
 
   def self.load_keychain
-    unless File.exist?(Awskeyring::PREFS_FILE) || prefs.empty?
-      warn "Config missing, run `#{$PROGRAM_NAME} initialise` to recreate."
+    unless File.exist?(Awskeyring::PREFS_FILE) && !prefs.empty?
+      warn "Config missing, run `#{File.basename($PROGRAM_NAME)} initialise` to recreate."
```

Comment:
Are they actually 'recreating'?

Reply:
Guessing `.empty?` is actually a Ruby helper for determining an empty object?

Reply:
If the file is missing or prefs is an empty hash you'd need to init.